src/module.h

```diff
@@ -10,6 +10,60 @@
 
 #include "RedisModulesSDK/redismodule.h"
 
+/// @brief Check if the key is allowed by the ACLs for the current user.
+/// @param ctx The redis module context.
+/// @param keyName The name of the key to check the ACLs for.
+/// @param permissionFlags The permissions to check for.
+/// @return true if the key is allowed by the ACLs, false otherwise.
+static inline __attribute__((always_inline)) bool CheckKeyIsAllowedByAcls(
```

Comment:
Should this be questionable, we could benchmark this, it might be too aggressive due to the amount of code.